give guidance to reduce scope of constants (#217)
uber-go/guide: src/var-scope.md

> # Reduce Scope of Variables and Constants

Author: This might mess up existing links?

If so, we can just update description and examples

Reviewer: Yeah let's not edit the heading if possible. I think the description is pretty clear.

</td></tr>
</tbody></table>

> Avoid making constants global unless they are used in multiple functions.

Reviewer: ```suggestion
Constants do not need to be global unless they are used in multiple functions or files.
```

> Constants do not need to be global unless they are used in multiple functions or files.

Reviewer 2: optional qualification:

```suggestion
Constants do not need to be global unless they are used in multiple functions or files
or are part of an external contract of the package.
```

<table>
<thead><tr><th>Bad</th><th>Good</th></tr></thead>
<tbody>
<tr><td>

```go
const (
  _defaultPort = 8080
  _defaultUser = "user"
)

func Bar() {
  fmt.Println("Default port", _defaultPort)
}
```

</td><td>

```go
func Bar() {
  const (
    defaultPort = 8080
    defaultUser = "user"
  )

>   fmt.Println("Default port", _defaultPort)

Reviewer 2: ```suggestion
  fmt.Println("Default port", defaultPort)
```
